aws/ami_test.go

```diff
@@ -0,0 +1,116 @@
+package aws
+
+import (
+	"testing"
+	"time"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/aws-nuke/util"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func createTestAMI(t *testing.T, session *session.Session, name string) ec2.Image {
+	svc := ec2.New(session)
+	instance := createTestEC2Instance(t, session, name, false)
```

Comment:
Should this `defer` the deletion of this instance?

Reply:
It does the defer on line 62. If the `defer` is done here, the EC2 instance could have been nuked before we get a chance to create the Image. The safest bet is to just clean up after the test function itself